presto-main/src/main/java/com/facebook/presto/transaction/TransactionManager.java

```diff
@@ -390,6 +390,12 @@ private synchronized CatalogMetadata getTransactionCatalogMetadata(ConnectorId c
             CatalogMetadata catalogMetadata = this.catalogMetadata.get(connectorId);
             if (catalogMetadata == null) {
                 Catalog catalog = catalogsByConnectorId.get(connectorId);
+                if (catalog == null) {
+                    // For rule tester, getConnectorId has never been called because the plan was generated outside this transaction.
```

Comment:
I am not sure if need to change the product because the testing infrastructure code. Maybe we should testing code instead to not have this issue?